parser-typechecker/src/Unison/Syntax/TermPrinter.hs

```diff
@@ -159,14 +162,12 @@ data DocLiteralContext
 -}
 
 pretty0 ::
-  forall v.
-  Var v =>
-  PrettyPrintEnv ->
+  forall v m.
+  MonadPretty v m =>
```

Comment:
The pretty-printer now uses a monad `m` which is constrained to be a `MonadReader` of the environment, including `PrettyPrintEnv` and the set of bound type variable names.

I decided to make this a constraint rather than a concrete class to make this easier to work with in the future, where the context may be something like `ReaderT PPEnv IO` or something. Avoids lifting and hoisting nonsense down the line.

Reply:
Yeah, there's a pretty good chance the pretty-printer will eventually read names directly from Sqlite on demand, so we'll need a pretty-print monad at that point anyways. Thanks for getting a good head-start on it 👍🏼